### PR TITLE
Fix t3c test duplication and upgrade to v4 (fix conflicts)

### PR DESCRIPTION
**Pull request by ericholguin**
Clone of #7167 

Differences:
- Fixes merge conflicts with the master branch
- As well as separates all of the CreateTest* and DeleteTest* functions into individual files instead of one large file (moved to `/lib/go-tc/totest` instead of `/traffic_ops/testing/api/v4/totest`

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Control Cache Config Integration Tests (`t3c`, formerly ORT)
- Traffic Control Health Client Integration Tests(tc-health-client)
- Traffic Ops Integration Tests

## What is the best way to verify this PR?

Run:
- Traffic Control Cache Config Integration Tests
- Traffic Control Health Client Integration Tests
- Traffic Ops Integration Tests

## PR submission checklist
- [x] This PR has tests 
- [ ] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by zrhoffman** [comment]
#7287 changes (or removes) 130 files, which is too large of an impact for a single PR. Would you please break #7287 up into some smaller PRs, each of which could hopefully be reviewed by a single person?

**Comment by ericholguin** [comment]
> #7287 changes (or removes) 130 files, which is too large of an impact for a single PR. Would you please break #7287 up into some smaller PRs, each of which could hopefully be reviewed by a single person?

I'm not sure that's necessary. The changes are cookie cutter changes, moving test creation/deletion functions to a shared folder used by both the TO integration and t3c integration tests, it touches a lot of files but these aren't complicated changes.

File: infrastructure/cdn-in-a-box/enroller/Dockerfile [hunk 1]
```diff
@@ -38,6 +38,7 @@ ARG ENROLLER_DEBUG_BUILD=false
 COPY ./lib/ /go/src/github.com/apache/trafficcontrol/lib/
 COPY ./go.mod ./go.sum /go/src/github.com/apache/trafficcontrol/
 COPY ./vendor/ /go/src/github.com/apache/trafficcontrol/vendor/
+COPY ./traffic_ops/testing/api/assert /go/src/github.com/apache/trafficcontrol/lib/go-util/assert
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/enroller/Dockerfile:41]: I get an error building CDN in a Box

```docker
 => ERROR [enroller-builder 4/9] COPY ./traffic_ops/testing/api/assert /go/src/github.com/apache/trafficcontrol/  0.0s
------
 > [enroller-builder 4/9] COPY ./traffic_ops/testing/api/assert /go/src/github.com/apache/trafficcontrol/lib/go-util/assert:
------
failed to solve: failed to compute cache key: failed to calculate checksum of ref moby::9jat5t3pk862i31sl1e3jrmhw: "/traffic_ops/testing/api/assert": not found
```

**Review by zrhoffman — CHANGES_REQUESTED** [review]

**Review by zrhoffman — APPROVED** [review]
Looks good, thanks!

**Pull request merged**